x/observer/keeper/ballot.go

```diff
@@ -58,10 +71,25 @@ func (k Keeper) GetAllBallots(ctx sdk.Context) (voters []*types.Ballot) {
 
 // AddBallotToList adds a ballot to the list of ballots for a given height.
 func (k Keeper) AddBallotToList(ctx sdk.Context, ballot types.Ballot) {
-	list, found := k.GetBallotList(ctx, ballot.BallotCreationHeight)
+	list, found := k.GetBallotListForHeight(ctx, ballot.BallotCreationHeight)
 	if !found {
 		list = types.BallotListForHeight{Height: ballot.BallotCreationHeight, BallotsIndexList: []string{}}
 	}
 	list.BallotsIndexList = append(list.BallotsIndexList, ballot.BallotIdentifier)
 	k.SetBallotList(ctx, &list)
 }
+
+// ClearMaturedBallots deletes all matured ballots and the list of ballots for a given height.
+// It also emits an event for each ballot deleted.
+func (k Keeper) ClearMaturedBallots(ctx sdk.Context, ballots []types.Ballot, maturityBlocks int64) {
```

Comment:
This method should not accept any input besides context. Current method signature allows deleting any `ballots []types.Ballot`

Reply:
We need the `maturityBlocks` parameter , as its from the emissions params 
`ballots` are passed as a param to reduce the number of reads to the store , as ballot are already fetched once here 
https://github.com/zeta-chain/zeta-node/blob/afaa312cdc815babff7ec59bc90cca36b45f0d0d/x/emissions/abci.go#L106-L114

The function name can be changed though ,cause it does more than just clear ballots , it deletes the ballot list as well